src/Contracts/Str.php

```diff
@@ -0,0 +1,46 @@
+<?php
```

Comment:
@naroga if we're using as objects, we cannot copy the variable without clone. Remember they are objects so if we do a toUpperCase() on the variable, it will have undesired effects if not immutable